.github/workflows/diff_report.yml

```diff
@@ -94,14 +95,25 @@ jobs:
          mv config.xml ./contribution/checkstyle-tester/
          mv project.properties ./contribution/checkstyle-tester/
          
-     - name: Generate report
+     - name: Generate basic difference report
+       if: github.event.comment.body == 'diff report'
        run: |
+         sudo apt install groovy
+         cd contribution/checkstyle-tester
          bash
          REF="origin/${{needs.parse_body.outputs.branch}}"
          REPO="../../checkstyle"
-         cd contribution/checkstyle-tester
-         sudo apt install groovy
          groovy diff.groovy -r $REPO -b master -p $REF -c config.xml -l project.properties
+         
+     - name: Generate basic single report
+       if: github.event.comment.body == 'single report'
+       run: |
+         sudo apt install groovy
```

Comment:
` sudo apt install groovy`
This can be moved to previous step "Relocate .xml and .properties ". And "Relocate .xml and .properties" can be changed to "Prepare environment" or something like this